metrics_utility/automation_controller_billing/dataframe_engine/db_dataframe_host_metric.py

```diff
@@ -34,6 +34,6 @@ def build_dataframe(self):
                 host_metric_concat = pd.concat([host_metric_concat, host_metric], ignore_index=True)
 
         if host_metric_concat is None:
```

Comment:
```suggestion
        if host_metric_concat is None or host_metric_concat.empty:
```

(and the others too - otherwise we can still return an empty dataframe without columns)